packages/eslint-plugin-jsx/rules/jsx-closing-bracket-location/jsx-closing-bracket-location.ts

```diff
@@ -185,25 +212,35 @@ export default {
       return indentation + spaces.join(' ')
     }
 
+    interface LastPropLocation {
+      column: number
+      firstLine: number
+      lastLine: number
+    }
+
+    type LastProp = LastPropLocation | undefined | Tree.JSXOpeningElement['attributes'][number]
+
+    type Tokens = ReturnType<typeof getTokensLocations>
```

Comment:
Can we have those types at the top level? Thanks

Reply:
That might need to extract `getTokensLocation` to the top level.

The type of `Tokens` is too complex to handwrite.